topi/python/topi/arm_cpu/conv2d.py

```diff
@@ -0,0 +1,515 @@
+    co, ci = cfg['tile_c'].apply(s, M, c)
+    xo, xi = cfg['tile_p'].apply(s, M, b)
+    s[M].reorder(eps, nu, xo, co, k, ci, xi)
+    cfg.define_annotate('ann_reduce', [ci], policy='try_unroll')
+    cfg.define_annotate('ann_spatial', [k, xi], policy='try_unroll_vec')
+    cfg['ann_reduce'].apply(s, M, [ci],
+                            axis_lens=[cfg['tile_c'].size[-1]],
+                            max_unroll=16,
+                            cfg=cfg)
+    cfg['ann_spatial'].apply(s, M, [k, xi])
+
+    # inverse transform
+    s[A].compute_inline()
+    k, b, vh, vw = s[Y].op.axis
+    r_eps, r_nu = s[Y].op.reduce_axis
+    s[Y].unroll(vh)
+    s[Y].unroll(vw)
+    s[Y].unroll(r_eps)
+    s[Y].unroll(r_nu)
+
+    # output
+    n, co, h, w = s[last].op.axis
+    co, coi = cfg['tile_k'].apply(s, last, co)
+    s[M].compute_at(s[last], co)
+    s[last].parallel(co)
+
+    MM = s.cache_read(M, 'global', [Y])
+    m = get_const_int(A.shape[1])
+    ho, wo, hi, wi = s[last].tile(h, w, m, m)
+    s[Y].compute_at(s[last], wo)
+    s[MM].compute_at(s[last], wo)
+
+    if output != last:
+        s[output].compute_inline()
+
+def _winograd_conv_arg_to_workload(data, kernel, strides, padding, layout, out_dtype, tile_size):
+    """convert argument to workload"""
+    K = 3
+    shape = get_const_tuple(kernel.shape)
+    alpha = tile_size + K - 1
+    if len(kernel.shape) == 4:
+        assert shape[2:] == (K, K)
+        CO, CI = shape[:2]
+    else:
+        assert shape[:2] == (alpha, alpha)
+        CO, CI, VCO = shape[2:]
+        CO *= VCO
+
+    raw_kernel = tvm.placeholder((CO, CI, K, K), dtype=kernel.dtype)
+    return ('conv2d', ) + autotvm.task.args_to_workload(
+        [data, raw_kernel, strides, padding, layout, out_dtype])
+
+
+@conv2d_winograd_without_weight_transform.register(['arm_cpu'])
+@autotvm.task.dispatcher
+def winograd_ww_config_dispatcher_(data, kernel, strides, padding, layout, out_dtype, tile_size):
+    return _winograd_conv_arg_to_workload(data, kernel, strides, padding, layout, out_dtype,
+                                          tile_size)
+
+
+@winograd_ww_config_dispatcher_.register(['winograd'])
+def decl_winograd_ww(cfg, data, kernel, strides, padding, layout, out_dtype, tile_size):
+    return _decl_winograd(cfg, data, kernel, strides, padding, layout, out_dtype,
+                          tile_size)
+
+
+@autotvm.task.register_topi_schedule(schedule_conv2d_winograd_without_weight_transform,
+                                     'arm_cpu', ['winograd'])
+def schedule_conv2d_winograd_without_weight_transform_(cfg, outs):
+    """TOPI schedule callback"""
+    s = tvm.create_schedule([x.op for x in outs])
+
+    def _callback(op):
+        if 'winograd_conv_output' in op.tag:
+            output = op.output(0)
+            _schedule_winograd(cfg, s, output, outs[0])
+
+    traverse_inline(s, outs[0].op, _callback)
+    return s
+
+
+@conv2d_alter_layout.register(["arm_cpu", "mali"])
```

Comment:
@FrozenGene I registered alter_layout for arm_cpu here. I didn't get any error.

Reply:
Got it.

Reply:
@FrozenGene  I added pre-tuned parameters for pynq board, which has a dual Cortex-A9 cpu.